File: src/main/java/ru/javarush/cryptoanalyzer/konovalov/controller/ActionContainer.java
package ru.javarush.cryptoanalyzer.konovalov.controller;


import ru.javarush.cryptoanalyzer.konovalov.Commands.*;
import ru.javarush.cryptoanalyzer.konovalov.data.Constants;
import ru.javarush.cryptoanalyzer.konovalov.exception.AppException;

public enum ActionContainer {
    ENCODE(new CaesarEncoder()),
    DECODE(new CaesarDecoder()),
    BRUTEFORCE(new BruteForceDecoder()),
    ANALYZE(new StatAnalyzeDecoder()),
    EXIT(new EXIT());

    private final Action ACTION;

    ActionContainer(Action action){
        this.ACTION = action;
    }

    public static Action getAction(String actionName) {
        try{
            ActionContainer value = ActionContainer.valueOf(actionName.toUpperCase());
            return value.ACTION;
        }
        catch( IllegalArgumentException e){
            String message = String.format(Action.NOT_FOUND_ACTION_FORMAT, actionName);
            throw new AppException(message,e);
        }
    }
}

File: src/main/kotlin/controller/ActionContainer.kt
package controller

import commands.Action
import commands.*
import data.*
import exception.AppException

enum class ActionContainer(action: Action) {
    ENCODE(CaesarEncoder()),
    DECODE(CaesarDecoder()),
    BRUTEFORCE(BruteForceDecoder()),
    ANALYZE(StatAnalyzeDecoder()),
    EXIT(EXIT());

    private val ACTION: Action = action

    companion object{
        @JvmStatic fun getAction(actionName: String): Action {
            try{
                val value = ActionContainer.valueOf(actionName.uppercase())
                return value.ACTION
            }
            catch (e: IllegalArgumentException){
            val message: String = String.format(Constants.NOT_FOUND_ACTION_FORMAT.name, actionName)
                throw AppException(message,e)
            }
        }
    }
}